src/TextInput.tsx

```diff
@@ -78,13 +87,12 @@ const TextInput = React.forwardRef<HTMLInputElement, TextInputInternalProps>(
       </TextInputWrapper>
     )
   }
-)
+) as PolymorphicForwardRefComponent<'input', TextInputProps>
```

Comment:
Using the polymorphic type utility from Radix here to implement correct types for the `as` prop.

Eventually, we should deprecate the `as` prop for TextInput. I can't think of a good use case for it now that we have the `Textarea` component